### PR TITLE
Cancel PRs automatically if new commits are pushed

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,6 +4,14 @@ on:
     branches:
       - main
   pull_request:
+
+concurrency:
+  # Taken from gradle/gradle
+  # On main, we don't want any jobs cancelled so the sha is used to name the group
+  # On PR branches, we cancel the job if new commits are pushed
+  group: ${{ (github.ref == 'refs/heads/main' ||  startsWith(github.ref, 'refs/tags') ) && format('contributor-pr-base-{0}', github.sha) || format('contributor-pr-{0}', github.ref) }}
```
Comment (dos65) [inline .github/workflows/ci.yml:12]: Just curious - have found an explanation for this kind of `if/else`?
I've tried but without success

**Review by dos65 — APPROVED** [review]
Lets' try! :rocket:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tgodzik) [reply .github/workflows/ci.yml:12]: It's just the typescript thing with. You can join `Boolean` and other value so `true && something` will be something and similar with `||`

**Comment by ckipp01** [comment]
This wasn't working as expected for me last night. I sort of assumed this mean you push a commit, and then you push another and it stops the first and starts the second. However,  in. my case it was the other way around. I second was halted _until_ the first was finished. Just to clarify, was is the exact intention here?

**Comment by tgodzik** [comment]
> This wasn't working as expected for me last night. I sort of assumed this mean you push a commit, and then you push another and it stops the first and starts the second. However, in. my case it was the other way around. I second was halted _until_ the first was finished. Just to clarify, was is the exact intention here?

Damn, it was supposed to stop the first one and run the second. It might be waiting for the other job to cancel though, but when I was testing it, everything was working as expected :/

**Comment by tgodzik** [comment]
It seems to have worked in https://github.com/scalameta/metals/actions/runs/1863084519 :thinking:

**Comment by ckipp01** [comment]
🤔 I have rebased but maybe it was a funky timing thing. I'll check again and see.

**Comment by tgodzik** [comment]
> Ahhh yea that was my bad. I was impatient laughing. When you first push up you see
> 
> And the little clock1 appears until it's finishing cancelling your other stuff. I didn't realize that's what it was doing and thought it was waiting for it to finish instead.

Ufff :sweat_smile:

**Pull request merged**